packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -322,14 +326,38 @@ export default class {
 
     xapi.xo.install()
 
-    await xapi.connect().then(
-      () => this.updateXenServer(id, { error: null }),
-      error => {
-        this.updateXenServer(id, { error: serializeError(error) })
+    let urlAfterRedirect
+    xapi.on('redirect', url => {
+      urlAfterRedirect = url
+    })
 
+    await xapi.connect().catch(
+      error => {
+        this.updateXenServer(id, { error: serializeError(error) })::ignoreErrors()
         throw error
       }
     )
+
+    if (urlAfterRedirect !== undefined) {
+      const servers = await this.getAllXenServers()
+      const serverExists = some(
+        servers,
+        server => isEqual(parseUrl(server.host), urlAfterRedirect)
+      )
+
+      if (!serverExists) {
+        return this.updateXenServer(id, { host: urlAfterRedirect.hostname, force: true, error: null })
```

Comment:
`hostname` is not the same as host, you might be losing the protocol and the port here…